coderd/autobuild/lifecycle_executor_test.go

```diff
@@ -64,50 +65,128 @@ func TestExecutorAutostartOK(t *testing.T) {
-			cwr.AutostartSchedule = ptr.Ref(sched.String())
-		})
-	)
-	// Given: workspace is stopped
-	workspace = coderdtest.MustTransitionWorkspace(t, client, workspace.ID, database.WorkspaceTransitionStart, database.WorkspaceTransitionStop)
-
-	// Given: the workspace template has been updated
-	orgs, err := client.OrganizationsByUser(ctx, workspace.OwnerID.String())
-	require.NoError(t, err)
-	require.Len(t, orgs, 1)
-
-	newVersion := coderdtest.UpdateTemplateVersion(t, client, orgs[0].ID, nil, workspace.TemplateID)
-	coderdtest.AwaitTemplateVersionJobCompleted(t, client, newVersion.ID)
-	require.NoError(t, client.UpdateActiveTemplateVersion(ctx, workspace.TemplateID, codersdk.UpdateActiveTemplateVersion{
-		ID: newVersion.ID,
-	}))
-
-	// When: the autobuild executor ticks after the scheduled time
-	go func() {
-		tickCh <- sched.Next(workspace.LatestBuild.CreatedAt)
-		close(tickCh)
-	}()
+	testCases := []struct {
+		name                 string
+		automaticUpdates     codersdk.AutomaticUpdates
+		compatibleParameters bool
+		expectStart          bool
+		expectUpdate         bool
+	}{
+		{
+			name:                 "Never",
+			automaticUpdates:     codersdk.AutomaticUpdatesNever,
+			compatibleParameters: true,
+			expectStart:          true,
+			expectUpdate:         false,
+		},
+		{
+			name:                 "Always_Compatible",
+			automaticUpdates:     codersdk.AutomaticUpdatesAlways,
+			compatibleParameters: true,
+			expectStart:          true,
+			expectUpdate:         true,
+		},
+		{
+			name:                 "Always_Incompatible",
+			automaticUpdates:     codersdk.AutomaticUpdatesAlways,
+			compatibleParameters: false,
+			expectStart:          false,
+			expectUpdate:         false,
+		},
+	}
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			var (
+				sched   = mustSchedule(t, "CRON_TZ=UTC 0 * * * *")
+				ctx     = context.Background()
+				err     error
+				tickCh  = make(chan time.Time)
+				statsCh = make(chan autobuild.Stats)
+				logger  = slogtest.Make(t, &slogtest.Options{IgnoreErrors: !tc.expectStart}).Leveled(slog.LevelDebug)
+				client  = coderdtest.New(t, &coderdtest.Options{
+					AutobuildTicker:          tickCh,
+					IncludeProvisionerDaemon: true,
+					AutobuildStats:           statsCh,
+					Logger:                   &logger,
+				})
+				// Given: we have a user with a workspace that has autostart enabled
+				workspace = mustProvisionWorkspace(t, client, func(cwr *codersdk.CreateWorkspaceRequest) {
+					cwr.AutostartSchedule = ptr.Ref(sched.String())
+					// Given: automatic updates from the test case
+					cwr.AutomaticUpdates = tc.automaticUpdates
+				})
+			)
+			// Given: workspace is stopped
+			workspace = coderdtest.MustTransitionWorkspace(
+				t, client, workspace.ID, database.WorkspaceTransitionStart, database.WorkspaceTransitionStop)
+
+			// Given: the workspace template has been updated
```

Comment:
This may have been my own copy-pasta from before

```suggestion
			// Given: an organization exists for the user
```